kernel/arch/i386/boot/bootparse.cpp

```diff
@@ -116,10 +123,166 @@ void px_parse_multiboot2(void *info)
+                                                                                    : "Kernel & Modules")));
+    }
+}
+
+void px_parse_stivale2(void *info)
+{
+    auto fixed = (struct stivale2_struct*)info;
+    for (uintptr_t page = ((uintptr_t)info & PAGE_ALIGN) + PAGE_SIZE;
+         page <= ((uintptr_t)info & PAGE_ALIGN);
+         page += PAGE_SIZE) {
+        px_rs232_printf("Mapping bootinfo at 0x%08x\n", page);
+        px_map_kernel_page(VADDR(page), page);
+    }
+    // Walk the list of tags in the header
+    auto tag = (struct stivale2_tag*)(fixed->tags);
+    while (tag)
+    {
+        // Map in each tag since Stivale2 doesn't give us a total size like Multiboot does.
+        uintptr_t page = ((uintptr_t)tag & PAGE_ALIGN);
+        px_map_kernel_page(VADDR(page), page);
+        // Follows the tag list order in stivale2.h
+        switch(tag->identifier)
+        {
+            case STIVALE2_STRUCT_TAG_CMDLINE_ID:
+            {
+                auto cmdline = (struct stivale2_struct_tag_cmdline*)tag;
+                px_rs232_printf("Stivale2 cmdline: '%s'\n", (const char *)cmdline->cmdline);
+                break;
+            }
+            case STIVALE2_STRUCT_TAG_MEMMAP_ID:
+            {
+                auto memmap = (struct stivale2_struct_tag_memmap*)tag;
+                px_rs232_printf("Stivale2 memory map:\n");
+                print_stivale2_mmap(memmap);
+                break;
+            }
+            case STIVALE2_STRUCT_TAG_FRAMEBUFFER_ID:
+            {
+                auto framebuffer = (struct stivale2_struct_tag_framebuffer*)tag;
+                px_rs232_printf("Stivale2 framebuffer:\n");
+                px_rs232_printf("  Address: 0x%08X", framebuffer->framebuffer_addr);
+                px_rs232_printf("  Resolution: %ix%ix%i\n",
+                    framebuffer->framebuffer_width,
+                    framebuffer->framebuffer_height,
+                    (framebuffer->framebuffer_bpp * 8));
+                break;
+            }
+            case STIVALE2_STRUCT_TAG_FB_MTRR_ID:
+            {
+                px_rs232_printf("  Framebuffer has MTRR\n");
+                break;
+            }
+            case STIVALE2_STRUCT_TAG_MODULES_ID:
+            {
+                auto modules = (struct stivale2_struct_tag_modules*)tag;
+                for (uint64_t i = 0; i < modules->module_count; i++) {
+                    stivale2_module mod = modules->modules[i];
+                    px_rs232_printf("Stivale2 module: %s\n  Module start: 0x%08x\n  Module end:   0x%08x\n",
+                        mod.string,
+                        mod.begin,
+                        mod.end
+                    );
+                }
+                break;
+            }
+            case STIVALE2_STRUCT_TAG_RSDP_ID:
+            {
+                auto rsdp = (struct stivale2_struct_tag_rsdp*)tag;
+                px_rs232_printf("ACPI RSDP: %08X\n", rsdp->rsdp);
```

Comment:
This is a really small thing, but would it be possible to share logic with Multiboot and print the RSDP info?

Reply:
We're only given a `uint64_t` for the rsdp and not a full struct. Is it worth going through the effort of converting the `uint64_t` into a Multiboot2 compliant struct just to print information?

Reply:
It's a pointer? Just cast it to a `struct RSDPDescriptor *`.

Reply:
I wouldn't assume that they're structured the same, but I'll give it a shot.

Reply:
Surprisingly it worked. Resolved in 19a72e6.